### ISSUE TITLE
Pagination logic working wrong in Takahe

### ISSUE THREAD

**Issue by alexpdp7** [ihabunek/toot#505]
(May apply to other servers too, not sure.)

If I scroll down until Toot paginates, I get an exception. The cause seems to be:

https://github.com/ihabunek/toot/blob/master/toot/api.py#L423

```
def _timeline_generator(app, user, path, params=None):
    while path:
        response = http.get(app, user, path, params)
        yield response.json()
        path = _get_next_path(response.headers)
```

If I put a breakpoint before the `response = ...` line, on the first load:

```
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/api.py(424)_timeline_generator()
-> response = http.get(app, user, path, params)
(Pdb) print(path, params)
/api/v1/timelines/home {'limit': 10}
```

This looks reasonable. However, when getting the next page:

```
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/api.py(424)_timeline_generator()
-> response = http.get(app, user, path, params)
(Pdb) print(path, params)
/api/v1/timelines/home?limit=10&max_id=374577459731194489 {'limit': 10}
(Pdb) s
--Call--
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(56)get()
-> def get(app, user, path, params=None, headers=None):
(Pdb) n
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(57)get()
-> url = app.base_url + path
(Pdb) n
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(59)get()
-> headers = headers or {}
(Pdb) n
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(60)get()
-> headers["Authorization"] = f"Bearer {user.access_token}"
(Pdb) n
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(62)get()
-> request = Request('GET', url, headers, params=params)
(Pdb) n
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(63)get()
-> response = send_request(request)
(Pdb) n
> /home/alex/.local/share/uv/tools/toot/lib/python3.11/site-packages/toot/http.py(65)get()
-> return process_response(response)
(Pdb) response.json()
{'error': 'invalid_input', 'error_details': [{'loc': ['limit'], 'msg': 'value is not a valid integer', 'type': 'type_error.integer'}]}
```

I think the problem is that both `path` and `params` contains a limit parameter, and somehow this borks the request.

If I patch the file to blow away `params`, then pagination works fine.

**Comment by alexpdp7** [comment]
https://github.com/alexpdp7/toot/commit/f079ebb638d84d4877f3ff6f66034ac8f99bc859

**Comment by ihabunek** [comment]
Thanks. Fixed and released in [0.47.0](https://github.com/ihabunek/toot/releases/tag/0.47.0).

**Comment by alexpdp7** [comment]
Sweet, thanks! I had reinstalled from Git and I was working fine. I have installed 0.47.0 and it's working too.

Great software!
